tests/test_json_schema.py

```diff
@@ -3334,139 +3334,143 @@ class Model(BaseModel):
-                                            'mapping': {
-                                                '0': '#/$defs/BlackCatWithHeight',
-                                                '1': '#/$defs/BlackCatWithWeight',
-                                            },
-                                            'propertyName': 'info',
-                                        },
-                                        'oneOf': [
-                                            {'$ref': '#/$defs/BlackCatWithHeight'},
-                                            {'$ref': '#/$defs/BlackCatWithWeight'},
-                                        ],
+                'oneOf': [
+                    {
+                        'oneOf': [
+                            {
+                                'oneOf': [
+                                    {'$ref': '#/$defs/BlackCatWithHeight'},
+                                    {'$ref': '#/$defs/BlackCatWithWeight'},
+                                ],
+                                'discriminator': {
+                                    'propertyName': 'info',
+                                    'mapping': {
+                                        'PetInfo.height': '#/$defs/BlackCatWithHeight',
+                                        'PetInfo.weight': '#/$defs/BlackCatWithWeight',
                                     },
-                                    'white': '#/$defs/WhiteCat',
                                 },
-                                'propertyName': 'color',
                             },
-                            'oneOf': [
-                                {
+                            {'$ref': '#/$defs/WhiteCat'},
+                        ],
+                        'discriminator': {
+                            'propertyName': 'color',
+                            'mapping': {
+                                'PetColor.black': {
+                                    'oneOf': [
+                                        {'$ref': '#/$defs/BlackCatWithHeight'},
+                                        {'$ref': '#/$defs/BlackCatWithWeight'},
+                                    ],
                                     'discriminator': {
+                                        'propertyName': 'info',
                                         'mapping': {
-                                            '0': '#/$defs/BlackCatWithHeight',
-                                            '1': '#/$defs/BlackCatWithWeight',
+                                            'PetInfo.height': '#/$defs/BlackCatWithHeight',
+                                            'PetInfo.weight': '#/$defs/BlackCatWithWeight',
                                         },
-                                        'propertyName': 'info',
                                     },
+                                },
+                                'PetColor.white': '#/$defs/WhiteCat',
```

Comment:
Same issue here, the key here should be `'white'` or it will break OpenAPI tooling making use of the discriminator functionality